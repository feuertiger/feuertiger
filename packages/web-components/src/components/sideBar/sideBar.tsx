import React from 'react';
import Link from 'next/link';
import styled, { css } from 'styled-components';
import {
    IconButton,
    Drawer,
    Divider,
    List,
    ListItem,
    ListItemIcon,
    ListItemText
} from '@material-ui/core';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import DashboardIcon from '@material-ui/icons/Dashboard';
import PeopleIcon from '@material-ui/icons/People';
import PeopleAltIcon from '@material-ui/icons/PeopleAlt';

const drawerWidth = 240;

const drawerClosed = ({ theme }) => css`
    overflow-x: hidden;
    width: ${theme.spacing(7)}px;
    ${theme.breakpoints.up('sm')} {
        width: ${theme.spacing(9)}px;
    }
`;

// eslint-disable-next-line
const StyledDrawer = styled(({ open, ...props }) => <Drawer {...props} />)`
    > div {
        position: relative;
        transition: ${({ open, theme: { transitions } }) =>
            transitions.create('width', {
                easing: transitions.easing.sharp,
                duration: open
                    ? transitions.duration.enteringScreen
                    : transitions.duration.leavingScreen
            })};
        white-space: nowrap;
        width: ${drawerWidth}px;
        z-index: 0;
        ${({ open, theme }) => (open ? '' : drawerClosed({ theme }))}
    }
`;

const StyledDiv = styled.div`
    display: flex;
    align-items: center;
    justify-content: flex-end;
    padding: 0 8px;
    ${({ theme: { mixins } }) => mixins.toolbar}
`;

export interface SideBarProps {
    open: boolean;
    handleDrawerClose: () => void;
}

export const mainListItems = (
    <div>
        <Link href="/">
            <ListItem button>
                <ListItemIcon>
                    <DashboardIcon />
                </ListItemIcon>
                <ListItemText primary="Dashboard" />
            </ListItem>
        </Link>
        <Link href="/member">
            <ListItem button>
                <ListItemIcon>
                    <PeopleAltIcon />
                </ListItemIcon>
                <ListItemText primary="Mitglieder" />
            </ListItem>
        </Link>
        <Link href="/exercices ">
            <ListItem button>
                <ListItemIcon>
                    <PeopleIcon />
                </ListItemIcon>
                <ListItemText primary="Ãœbungen" />
            </ListItem>
        </Link>
    </div>
);

export const secondaryListItems = <div />;

export const SideBar = ({ open, handleDrawerClose }: SideBarProps) => (
    <StyledDrawer variant="permanent" open={open}>
        <StyledDiv>
            <IconButton onClick={handleDrawerClose}>
                <ChevronLeftIcon />
            </IconButton>
        </StyledDiv>
        <Divider />
        <List>{mainListItems}</List>
        <Divider />
        <List>{secondaryListItems}</List>
    </StyledDrawer>
);
