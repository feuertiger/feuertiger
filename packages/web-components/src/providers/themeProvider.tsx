/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';
import { ThemeProvider as StyledComponentThemeProvider } from 'styled-components';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import DateFnsUtils from '@date-io/date-fns';

const theme = createMuiTheme({
    palette: {
        type: 'dark',
        primary: {
            main: '#c90024'
        }
    }
});

export interface ThemeProviderProps {
    children: JSX.Element | (false | JSX.Element)[];
}

export class ThemeProvider extends React.Component<ThemeProviderProps> {
    componentDidMount() {
        const jssStyles = document.querySelector('#jss-server-side');
        if (jssStyles && jssStyles.parentNode) {
            jssStyles.parentNode.removeChild(jssStyles);
        }
    }

    render() {
        const { children } = this.props;

        return (
            <StyledComponentThemeProvider theme={theme}>
                <MuiThemeProvider theme={theme}>
                    <MuiPickersUtilsProvider utils={DateFnsUtils}>
                        <>{children}</>
                    </MuiPickersUtilsProvider>
                </MuiThemeProvider>
            </StyledComponentThemeProvider>
        );
    }
}
