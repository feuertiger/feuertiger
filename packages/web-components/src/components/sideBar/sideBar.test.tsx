import { test, describe, expect } from '@jest/globals';
import { SideBar } from './sideBar';
import { ThemeProvider } from '../../providers/themeProvider';
import { shallow } from 'enzyme';

describe('Test side bar', () => {
    test('should create side bar', () => {
        const component = shallow(
            <SideBar open={false} handleDrawerClose={() => {}} />,
            {
                wrappingComponent: ThemeProvider
            }
        );
        expect(component).toMatchSnapshot();
    });
    test('should be open', () => {
        const component = shallow(
            <SideBar open={true} handleDrawerClose={() => {}} />,
            {
                wrappingComponent: ThemeProvider
            }
        );
        expect(component).toMatchSnapshot();
    });
    test('should be closed', () => {
        const component = shallow(
            <SideBar open={false} handleDrawerClose={() => {}} />,
            {
                wrappingComponent: ThemeProvider
            }
        );
        expect(component).toMatchSnapshot();
    });
    test('should trigger handleDrawerClose', () => {
        let triggerdClose = false;
        const component = shallow(
            <SideBar
                open={true}
                handleDrawerClose={() => {
                    triggerdClose = true;
                }}
            />,
            {
                wrappingComponent: ThemeProvider
            }
        );
        const closeButton = component.find('#sidebar-close-button');
        expect(closeButton).toBeTruthy();
        closeButton.simulate('click');
        expect(triggerdClose).toBeTruthy();
    });
});
