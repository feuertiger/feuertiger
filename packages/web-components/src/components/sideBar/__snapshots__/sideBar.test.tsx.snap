// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test side bar should be closed 1`] = `
<sideBar__StyledDrawer
  open={false}
  variant="permanent"
>
  <sideBar__StyledDiv>
    <WithStyles(ForwardRef(IconButton))
      id="sidebar-close-button"
      onClick={[Function]}
    >
      <ChevronLeftIcon />
    </WithStyles(ForwardRef(IconButton))>
  </sideBar__StyledDiv>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <div>
      <Link
        href="/"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DashboardIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Dashboard"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
      <Link
        href="/member"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <PeopleAltIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Mitglieder"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
      <Link
        href="/exercices "
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <PeopleIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Übungen"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
    </div>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <div />
  </WithStyles(ForwardRef(List))>
</sideBar__StyledDrawer>
`;

exports[`Test side bar should be open 1`] = `
<sideBar__StyledDrawer
  open={true}
  variant="permanent"
>
  <sideBar__StyledDiv>
    <WithStyles(ForwardRef(IconButton))
      id="sidebar-close-button"
      onClick={[Function]}
    >
      <ChevronLeftIcon />
    </WithStyles(ForwardRef(IconButton))>
  </sideBar__StyledDiv>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <div>
      <Link
        href="/"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DashboardIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Dashboard"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
      <Link
        href="/member"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <PeopleAltIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Mitglieder"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
      <Link
        href="/exercices "
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <PeopleIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Übungen"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
    </div>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <div />
  </WithStyles(ForwardRef(List))>
</sideBar__StyledDrawer>
`;

exports[`Test side bar should create side bar 1`] = `
<sideBar__StyledDrawer
  open={false}
  variant="permanent"
>
  <sideBar__StyledDiv>
    <WithStyles(ForwardRef(IconButton))
      id="sidebar-close-button"
      onClick={[Function]}
    >
      <ChevronLeftIcon />
    </WithStyles(ForwardRef(IconButton))>
  </sideBar__StyledDiv>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <div>
      <Link
        href="/"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DashboardIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Dashboard"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
      <Link
        href="/member"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <PeopleAltIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Mitglieder"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
      <Link
        href="/exercices "
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <PeopleIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Übungen"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
    </div>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <div />
  </WithStyles(ForwardRef(List))>
</sideBar__StyledDrawer>
`;
